common/src/main/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsManager.kt

```diff
@@ -69,14 +72,41 @@ class DiagnosticsManager(
 
     fun trackEvent(diagnosticsEvent: DiagnosticsEvent) {
         enqueue {
-            val anonymizedEvent = diagnosticsAnonymizer.anonymizeEventIfNeeded(diagnosticsEvent)
-            // WIP: Check that file size is not above certain limit. If it is, delete.
-            verboseLog("Tracking diagnostics event: $anonymizedEvent")
-            try {
-                diagnosticsFileHelper.appendEventToDiagnosticsFile(anonymizedEvent)
-            } catch (e: IOException) {
-                verboseLog("Error tracking diagnostics event: $e")
-            }
+            trackEventInCurrentThread(diagnosticsEvent)
+        }
+    }
+
+    private fun getEventsToSync(): List<JSONObject> {
```

Comment:
does this method need to be synchronized?

Reply:
Hmm I would say with our current calls the chances of a race condition are 0 but for the sake of completeness, you're correct 👍

Reply:
Actually, I've been thinking that while we can make this class support multi-threading, considering we are using the same thread for everything related to diagnostics that would complicate the class without need. I improved the docstring for `DiagnosticsManager` but lmk if you still think we should future-proof it against multi-threading.

Reply:
YAGNI 👍🏻 sometimes the best way to make things thread safe is to just document that a type can't be used from multiple threads and deal with that outside of the class, keeps things much simpler.